google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java

```diff
@@ -2589,6 +2590,70 @@ public void testCopyJob() throws InterruptedException, TimeoutException {
     assertTrue(remoteTable.delete());
   }
 
+  @Test
+  public void testSnapshotTableCopyJob() throws InterruptedException {
+    String sourceTableName = "test_copy_job_base_table";
+    // this creates a snapshot table at specified snapshotTime
+    String snapshotTableName = String.format("test_snapshot_table");
+    // Create source table
+    TableId sourceTableId = TableId.of(DATASET, sourceTableName);
+    StandardTableDefinition tableDefinition = StandardTableDefinition.of(TABLE_SCHEMA);
+    TableInfo tableInfo = TableInfo.of(sourceTableId, tableDefinition);
+    Table createdTable = bigquery.create(tableInfo);
```

Comment:
I'd consider creating a table with some data in it; as is this is an empty table, e.g. with a CTAS or instantiating from a load job.

Reply:
Yes, good idea. Did this w CTAS statement.